packages/nest/src/services/notifire.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { Notifire } from '@notifire/core';
+import { NotifireOptions } from '../interfaces';
+import { NOTIFIRE_OPTIONS } from '../helpers';
+
+@Injectable()
+export class NotifireService {
+  private notifire: any;
+  private readonly providers: Array<any>;
+  private readonly templates: Array<any>;
+
+  constructor(
+    @Inject(NOTIFIRE_OPTIONS) private _NotifireOptions: NotifireOptions
+  ) {
+    this.notifire = new Notifire();
+    this.providers = this._NotifireOptions.providers;
+    this.templates = this._NotifireOptions.templates;
+  }
+
+  get() {
+    if (!this.notifire) {
+      this.notifire = new Notifire();
+    }
+    return this.notifire;
+  }
+
+  async registerProviders(): Promise<any> {
+    const providerPromises = [];
+
+    for (let i = 0; i < this.providers.length; i += 1) {
```

Comment:
Can be simplified to:
```suggestion
    for (const provider of this.providers.length) {
```